core/db/migrate/20150314013438_add_missing_indexes.rb

```diff
@@ -0,0 +1,27 @@
+class AddMissingIndexes < ActiveRecord::Migration
+  def change
+    add_index   :spree_promotion_rules_users, 
+                        [:user_id, :promotion_rule_id], 
+                        name: 'index_promotion_rules_users_on_user_id_and_promotion_rule_id'
+    add_index   :spree_products_promotion_rules, 
+                        [:promotion_rule_id, :product_id], 
+                        name: 'index_products_promotion_rules_on_promotion_rule_and_product'
+    add_index :spree_orders, :canceler_id
+    add_index :spree_orders, :store_id
+    add_index :spree_orders_promotions, [:promotion_id, :order_id]
+    add_index :spree_properties_prototypes, :prototype_id
+    add_index   :spree_properties_prototypes, 
+                        [:prototype_id, :property_id], 
+                        name:  'index_properties_prototypes_on_prototype_and_property'
```

Comment:
My non-PostgreSQL is not great, but I know that in PostgreSQL it can use a multi-column index for a lookup on a single component of the index.

For example, if a multi-column index exists on `prototype_id` and `property_id` and you do a query just using `prototype_id`, PostgreSQL will use the index. It's most efficient when using the left-most column in the multi-column index.

If other databases behave similarly. The additional index:

```
add_index :spree_orders_promotions, [:promotion_id, :order_id]
```

would not be needed.

Some of the other multi-column indexes specified in this migration should also be considered if they are appropriate, and if there is an existing index that should be removed in favour of one of these.